internal/pkg/utils/testhelper/runner/runner.go

```diff
@@ -0,0 +1,142 @@
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/testproject"
+)
+
+const testTimeout = 3 * time.Minute
+
+type Runner struct {
+	t          *testing.T
+	tempDir    string
+	testsDir   string
+	workingDir string
+}
+
+func NewRunner(t *testing.T, testsDir string) *Runner {
+	t.Helper()
+
+	workingDir := filesystem.Join(testsDir, ".out")
+	assert.NoError(t, os.RemoveAll(workingDir))
+	assert.NoError(t, os.MkdirAll(workingDir, 0o755))
+
+	return &Runner{t: t, testsDir: testsDir, workingDir: workingDir, tempDir: t.TempDir()}
+}
+
+func (r *Runner) NewTest(t *testing.T, testDirName string) (*Test, context.CancelFunc) {
+	t.Helper()
+
+	testDir := filepath.Join(r.testsDir, testDirName)
+	workingDir := filepath.Join(r.workingDir, testDirName)
+
+	assert.NoError(t, os.RemoveAll(workingDir))
+	assert.NoError(t, os.MkdirAll(workingDir, 0o755))
+	assert.NoError(t, os.Chdir(workingDir))
+
+	testDirFS, err := aferofs.NewLocalFs(testDir)
+	assert.NoError(t, err)
+	workingDirFS, err := aferofs.NewLocalFs(workingDir)
+	assert.NoError(t, err)
+
+	project := testproject.GetTestProjectForTest(t)
+
+	// Create context with timeout.
+	// Acquiring a test project and setting it up is not part of the timeout.
+	ctx, cancelFn := context.WithTimeout(context.Background(), testTimeout)
+
+	// Create ENV provider
+	envProvider := storageenv.CreateStorageEnvTicketProvider(ctx, project.KeboolaProjectAPI(), project.Env())
+
+	envMap := project.Env()
+	// Enable templates private beta in tests
+	envMap.Set(`KBC_TEMPLATES_PRIVATE_BETA`, `true`)
+	// Disable version check
+	envMap.Set(`KBC_VERSION_CHECK`, `false`)
+	envMap.Set("TEST_KBC_PROJECT_ID_8DIG", fmt.Sprintf("%08d", cast.ToInt(envMap.Get("TEST_KBC_PROJECT_ID"))))
+
+	return &Test{
+		Runner:       *r,
+		ctx:          ctx,
+		env:          envMap,
+		envProvider:  envProvider,
+		project:      project,
+		t:            t,
+		testDir:      testDir,
+		testDirFS:    testDirFS,
+		workingDir:   workingDir,
+		workingDirFS: workingDirFS,
+	}, cancelFn
+}
+
+// ForEachTest loops through all dirs within `runner.testsDir` and runs the test in it.
+func (r *Runner) ForEachTest(runFn func(test *Test)) {
+	r.t.Helper()
+
+	// Run test for each directory
+	for _, testDirName := range testhelper.GetTestDirs(r.t, r.testsDir) {
+		testName := testDirName
+		r.t.Run(testName, func(t *testing.T) {
+			t.Parallel()
+
+			test, cancelFn := r.NewTest(t, testName)
+			defer cancelFn()
+			runFn(test)
+		})
+	}
+}
+
+// CompileBinary compiles a binary used in the test by running a make command.
+func (r *Runner) CompileBinary(
+	cmdDir string,
+	binaryName string,
+	binaryPathEnvName string,
+	makeCommand string,
+) string {
+	r.t.Helper()
+
+	binaryPath := filesystem.Join(r.tempDir, "/"+binaryName)
+	if runtime.GOOS == "windows" {
+		binaryPath += `.exe`
+	}
+
+	// Envs
+	envs, err := env.FromOs()
+	assert.NoError(r.t, err)
+	envs.Set(binaryPathEnvName, binaryPath)
```

Comment:
Tohle se mi moc nelíbí. Každý ten make command potřebuje env var s cestou k binárce s jiným názvem, ale nechal bych to asi na další iteraci. Možná by se to celý dalo upravit víc od podlahy už v těch make commandech, nebo to udělat úplně mimo make. (Případně přijímám návrhy. 😃)

Reply:
Potrebuje to `PATH + GO_*` envs pre kompilaciu.

Reply:
Alebo sa ti nepaci, ze ta path je cez ENV? 
To sa do `make` nedaju specifikovat argumenty, ide to iba cez ENV.

Reply:
Ne to chápu, jen že tady ta proměnná `binaryPathEnvName` je taková těžkopádná. (Resp. že tady ta `CompileBinary()` metoda má tak nějak moc parametrů mi přijde. 😄) Ale jak říkám, to je spíš prostor do budoucna.

Reply:
Ak sa ti to velmi nepaci, tak by to mohlo byt samostatne v kazdom balicku, kedze kazda service sa kompiluje trocha inak. Ale neriesil by som to.